usaspending_api/etl/elasticsearch_loader_helpers/index_config.py

```diff
@@ -94,10 +94,32 @@ def swap_aliases(client, config):
 
     try:
         if old_indexes:
-            client.indices.delete(index=old_indexes, ignore_unavailable=False)
-            logger.info(format_log(f"Deleted index(es) '{old_indexes}'", action="ES Alias"))
+            max_wait_time = 15  # in minutes
+            start_wait_time = time.time()
+            is_snapshot_conflict = is_snapshot_running(client, old_indexes)
+            if is_snapshot_conflict:
+                logger.info(
+                    format_log(
+                        f"Snapshot in-progress prevents delete; waiting up to {max_wait_time} minutes",
+                        action="ES Alias",
+                    )
+                )
+            while (time.time() - start_wait_time) < (max_wait_time * 60) and is_snapshot_conflict:
+                logger.info(format_log("Waiting while snapshot is in-progress", action="ES Alias"))
+                time.sleep(90)
+                is_snapshot_conflict = is_snapshot_running(client, old_indexes)
+            if is_snapshot_conflict:
+                config["pipeline_exit_code"] = 1
```

Comment:
Did we want this to be an unstable pipeline status? The pattern for that in other scripts is `3` since 1 is the typical error status for everything

Reply:
That is the intended behavior and will make the change.